### PR TITLE
fix: replace node buffers with uint8arrays

### PR DESCRIPTION
**Pull request by achingbrain**
Upgrades all deps and replaces all use of node Buffers with Uint8Arrays

Depends on:

- [x] https://github.com/NodeFactoryIo/js-libp2p-noise/pull/70
- [x] https://github.com/libp2p/js-libp2p-interfaces/pull/62
- [x] https://github.com/libp2p/interop/pull/42

BREAKING CHANGES:

- All deps used by this module now use Uint8Arrays in place of node Buffers

### CODE REVIEW

**Comment by achingbrain** [comment]
This is using a temporary fork of libp2p-noise that has the changes from NodeFactoryIo/js-libp2p-noise#70 applied and the typescript compiled to js.

**Comment by jacobheun** [comment]
Looks like the browser tests are failing

File: test/keychain/cms-interop.spec.js [hunk 1]
```diff
@@ -58,7 +50,7 @@ cn4oisOvxCprs4aM9UVjtZTCjfyNpX8UWwT1W3rySV+KQNhxuMy3RzmL
     expect(key.id).to.equal(aliceKid)
   })
 
-  it('decrypts node-forge example', async () => {
+  it.only('decrypts node-forge example', async () => {
```
Comment (vasco-santos) [inline test/keychain/cms-interop.spec.js:53]:
```suggestion
  it('decrypts node-forge example', async () => {
```

Comment (achingbrain) [reply test/keychain/cms-interop.spec.js]: No wonder the tests were passing!

**Review by vasco-santos — APPROVED** [review]
LGTM!
pending the interface release

**Comment by jacobheun** [comment]
Ci is failing due to interop, this is a release order issue with the daemon. We can resolve this before release.

**Pull request merged**